portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -92,4 +106,18 @@ private String getParameter(HttpServletRequest request, String name, String defa
     }
     return value;
   }
+
+  /**
+   * Inner class for Comments.
+   */
+  class Comment {
+    private String text;
+    private Date date;
+
+    public Comment(String text, Date date) {
+      this.text = text;
+      this.date = date;
+    }
+  }
+
```

Comment:
This is a great use of an inner class!

Something I'd suggest here would be to move some of the Comment <-> 'Entity' logic into this class, eg: by adding a method which returns an Entity representing this Comment and a method which takes and Entity and produces the corresponding comment (consider a static method for the latter). Doing so keeps the nitty gritty logic of how to store and recover Comments encapsulated within the Comment class, and from the exterior you get a cleaner API where you can simply say "make this into an entity that i can store" and you don't need to worry about exactly which fields are set or what they're called.

This relates to my comment on your last CL since the Comment class can now encapsulate all the special key values ("Comment", "timestamp", "date", etc) in one place.

Final though, this touches on a common task in cs of serialization and parsing. These tasks tend to have a lot of complexity and there's lot of approaches out there for how to do this well, so feel free to ask any questions you have / do additional reading!

Reply:
That makes sense, thanks for the review! I've continued to expand on using the Comment class in later commits (not on this CL), and I just added a method within the Comment class to create an Entity, to clean up the logic flow and keep everything encapsulated (as you said).